go-controller/pkg/metrics/node.go

```diff
@@ -61,6 +61,9 @@ func RegisterNodeMetrics() {
 			func() float64 { return 1 },
 		))
 		registerWorkqueueMetrics(MetricOvnkubeNamespace, MetricOvnkubeSubsystemNode)
-		prometheus.MustRegister(MetricResourceRetryFailuresCount)
+		// Don't register twice when both NetworkControllerManager and Node modes are true
```

Comment:
What about putting these registers inside the Do once above?

Reply:
@martinkennelly  thanks for the review!
so I should be able to remove the ovnkube mode check by moving the metric registration into Once.Do, right? the Once.Do guarantees the metric will not be registered if already done in master mode?

Reply:
and by master mode you mean NCM + CM in a single process right. How are we registering node metrics in that mode?

Reply:
Yes, master mode means the NCM and CM.
This metric is not node specific, it is also registered in the RegisterMasterFunctional call: https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/metrics/master.go#L432.  RegisterMasterFunctional is in a different code path, not reuse the same Once.Do function, will it still guarantee the metric be registered once? From a quick look at the Once.Do doc, it doesn't seem so.

Reply:
I will give it a try.